### PR TITLE
DOC: updating the `indicator` wording in `merge` doc

### LINKED ISSUE #34480: DOC: `merge` doc for `indicator` parameter can be improved
#### Location of the documentation

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.merge.html#pandas.DataFrame.merge

#### Documentation problem

The documentation is correct but hard to read, currently:

```... If string, column with information on source of each row will be added to output DataFrame, and column will be named value of string. Information column is Categorical-type and ...```

#### Suggested fix for documentation

I believe the following paragraph will be easier to read:

```
If True, adds a column to the output DataFrame called “_merge” with information on 
the source of each row. The column can be given a different name by providing a 
string argument. The column will have a Categorical type with the value of “left_only” 
for observations whose merge key only appears in ‘left’ DataFrame, “right_only” for 
observations whose merge key only appears in ‘right’ DataFrame, and “both” if 
the observation’s merge key is found in both.
```

### PR DESCRIPTION
**Pull request by nnick14**
- [ ] closes #34480 
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/frame.py [hunk 1]
```diff
@@ -233,14 +233,13 @@
...
-    output DataFrame, and column will be named value of string.
-    Information column is Categorical-type and takes on a value of "left_only"
-    for observations whose merge key only appears in 'left' DataFrame,
-    "right_only" for observations whose merge key only appears in 'right'
-    DataFrame, and "both" if the observation's merge key is found in both.
+    If True, adds a column to the output DataFrame called "_merge" with
+    information on the source of each row. The column can be given a different
+    name by providing a string argument. The column will have a Categorical
+    type with the value of "left_only" for observations whose merge key only
+    appears in 'left' DataFrame, "right_only" for observations whose merge key
```
Comment (dsaxton) [inline pandas/core/frame.py:240]: 'left' -> the left, 'right' -> the right might sound / look better

Comment (nnick14) [reply pandas/core/frame.py]: Just adjusted the wording. Let me know if that looks good, or if I should add anything else.

Comment (dsaxton) [reply pandas/core/frame.py]: I meant replacing `'left'` with `the left` and similarly for `'right'` (i.e., "whose merge key only appears in the left DataFrame...")

Comment (nnick14) [reply pandas/core/frame.py]: Sorry, I misunderstood. It should be all set now.

**Review by mroeschke — APPROVED** [review]

**Review by WillAyd — APPROVED** [review]

**Comment by WillAyd** [comment]
Thanks @nnick14

**Pull request merged**